File: cpp/questions/minimumDifference.cpp
class Solution {
public:
    int minimumDifference(vector<int>& nums, int k) {
        int ans = INT_MAX;
        std::sort(nums.begin(), nums.end());
        for (int p=k-1; p<nums.size(); p++) {
            int curr_diff = nums[p] - nums[p-k+1];
            if (curr_diff < ans) {
                ans = curr_diff;
            }
        }
        return ans;
    }
};

File: javascript/questions/minimumDifference.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var minimumDifference = function(nums, k) {
    nums.sort((a, b) => a - b);
    const n = nums.length;
    let ans = Infinity;
    for (let p=k-1; p<n; p++) {
        const curr_diff = nums[p] - nums[p-k+1];
        ans = Math.min(ans, curr_diff);
    }
    return ans;
};
